Index: repy/safe.py
===================================================================
--- repy/safe.py	(revision 5111)
+++ repy/safe.py	(working copy)
@@ -74,6 +74,9 @@
 import platform # This is for detecting Nokia tablets
 import __builtin__
 
+# Needed for ticket #1049
+subprocess.getattr = getattr
+
 # Armon: This is how long we will wait for the external process
 # to validate the safety of the user code before we timeout, 
 # and exit with an exception
