Index: SeattleOnAndroid/src/com/seattleonandroid/ScriptActivity.java
===================================================================
--- SeattleOnAndroid/src/com/seattleonandroid/ScriptActivity.java	(revision 5948)
+++ SeattleOnAndroid/src/com/seattleonandroid/ScriptActivity.java	(working copy)
@@ -239,8 +239,18 @@
 	}
 
 	// Back button was pressed by the user
+	@Override
 	public void onBackPressed() {
-		goBack();
+		if(currentContentView == R.layout.logmenuview){
+			showFrontendLayout();
+		} else if(currentContentView == R.layout.logfileview){
+			showAvailableLogListing();
+		} else if(currentContentView == R.layout.about){
+			showFrontendLayout();
+		} else{
+			// Let android handle it.
+			super.onBackPressed();
+		}
 	}
 
 	// Get a listing of the (hopefully) seattle specific log files in the directory 
@@ -359,7 +369,7 @@
 				return true;
 			case R.id.back:
 				// Back button
-				goBack();
+				onBackPressed();
 				return true;
 			case R.id.refresh:
 				// Refresh
@@ -389,17 +399,6 @@
 		}
 	}
 
-	// Back button pressed
-	private void goBack(){
-		if(currentContentView == R.layout.logmenuview){
-			showFrontendLayout();
-		} else if(currentContentView == R.layout.logfileview){
-			showAvailableLogListing();
-		} else if(currentContentView == R.layout.about){
-			showFrontendLayout();
-		}
-	}
-
 	// Show the main (seattle already installed) layout
 	private void showMainLayout() {
 		setContentView(R.layout.main);
