Index: C:/Users/Leonard/Dropbox/Seattle/seattle/branches/repy_v2/preparetest.py
===================================================================
--- C:/Users/Leonard/Dropbox/Seattle/seattle/branches/repy_v2/preparetest.py	(revision 7003)
+++ C:/Users/Leonard/Dropbox/Seattle/seattle/branches/repy_v2/preparetest.py	(working copy)
@@ -74,6 +74,42 @@
       shutil.copyfile(file_path,target +"/"+os.path.basename(file_path))
 
 
+def copy_tree_to_target(source, target, ignore=None):
+  """
+  Copies a directory to the target destination.
+
+  If you pass a string for ignore, then subdirectories that contain the ignore
+  string will not be copied over (as well as the files they contain).
+  """
+
+  full_source_path = os.path.abspath(source)
+  full_target_path = os.path.abspath(target)
+
+  for root, directories, filenames in os.walk(source):
+    # Relative path is needed to build the absolute target path.
+
+    # If we leave a leading directory separator in the relative folder
+    # path, then attempts to join it will cause the relative folder path
+    # to be treated as an absolute path.
+    relative_folder_path = os.path.abspath(root)[len(full_source_path):].lstrip(os.sep)
+
+    # If the ignore string is in the relative path, skip this directory.
+    if ignore and ignore in relative_folder_path:
+      continue
+
+    # Attempts to copy over a file when the containing directories above it do not
+    # exist will trigger an exception.
+    full_target_subdir_path = os.path.join(full_target_path, relative_folder_path)
+    if not os.path.isdir(full_target_subdir_path):
+      os.makedirs(full_target_subdir_path)
+
+    for name in filenames:
+      relative_path = os.path.join(relative_folder_path, name)
+      shutil.copyfile(
+        os.path.join(full_source_path, relative_path),
+        os.path.join(full_target_path, relative_path))
+
+
 # Run the .mix files in current directory through the preprocessor 
 # script_path specifies the name of the preprocessor script
 # The preprocessor script must be in the working directory
@@ -218,6 +254,8 @@
   copy_to_target("seattlelib/textops.py", os.path.join(target_dir, "repyV2"))
   copy_to_target("nodemanager/servicelogger.py", os.path.join(target_dir, "repyV2"))
   copy_to_target("seash/*", target_dir)
+  copy_tree_to_target("seash/pyreadline/", os.path.join(target_dir, 'pyreadline/'), ignore=".svn")
+  copy_tree_to_target("seash/modules/", os.path.join(target_dir, 'modules/'), ignore=".svn")
   copy_to_target("affix/*", target_dir)
   #copy_to_target("shims/proxy/*", target_dir)
   copy_to_target("softwareupdater/*", target_dir)
@@ -240,6 +278,7 @@
     copy_to_target("nodemanager/tests/*", target_dir)
     copy_to_target("portability/tests/*", target_dir)
     copy_to_target("seash/tests/*", target_dir)
+    copy_tree_to_target("seash/tests/modules/", os.path.join(target_dir, 'modules/'), ignore=".svn")
     copy_to_target("seattlelib/tests/*", target_dir)
     #copy_to_target("keydaemon/tests/*", target_dir)
     copy_to_target("shims/tests/*", target_dir)
