--- C:\Users\Leonard\Dropbox\Seattle\seattle\trunk\seash\seash.py	Mon Sep 16 22:43:19 2013
+++ C:\Users\Leonard\Dropbox\Seattle\seattle\branches\repy_v2\seash\seash.py	Mon Feb  3 15:36:11 2014
@@ -50,6 +50,12 @@
 
 # simple client.   A better test client (but nothing like what a real client
 # would be)
+
+# Restores the original python built-in type() to certain built-in libraries
+# as repyportability destroys it.
+
+# Needed for parsing user commands and executing command functions
+import seash_dictionary
 
 ### Integration fix here...
 from repyportability import *
@@ -93,9 +99,6 @@
 # Don't hide mac readline.so if we didn't expose it
 if rename_readline_so_file:
   os.rename(EXPOSED_READLINE_SO_FN, HIDDEN_READLINE_SO_FN)
-
-# Needed for parsing user commands and executing command functions
-import seash_dictionary
 
 # Used for re-enabling modules on the last run
 import seash_modules